staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -544,19 +544,37 @@ redirectLoop:
 	return connToReturn, rawResponse.Bytes(), nil
 }
 
-// CloneRequest creates a shallow copy of the request along with a deep copy of the Headers.
+// CloneRequest creates a shallow copy of the request along with a deep copy of the Headers and the URL.
 func CloneRequest(req *http.Request) *http.Request {
 	r := new(http.Request)
 
 	// shallow clone
 	*r = *req
 
+	// deep copy URL
+	r.URL = CloneURL(req.URL)
+
 	// deep copy headers
 	r.Header = CloneHeader(req.Header)
 
 	return r
 }
 
+// CloneURL creates a deep copy of an URL
+// https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/net/http/clone.go#L22
+func CloneURL(u *url.URL) *url.URL {
+	if u == nil {
+		return nil
+	}
+	u2 := new(url.URL)
+	*u2 = *u
+	if u.User != nil {
+		u2.User = new(url.Userinfo)
+		*u2.User = *u.User
+	}
+	return u2
+}
```

Comment:
would reparsing be more compatible? This here could break, couldn't it?

Reply:
This is how the stdlin implements it, how can it break?
The only field that can be mutated is userinfo
https://github.com/golang/go/issues/41733